internal/util/function/highlight/semantic_highlight.go

```diff
@@ -0,0 +1,146 @@
+/*
+ * # Licensed to the LF AI & Data foundation under one
+ * # or more contributor license agreements. See the NOTICE file
+ * # distributed with this work for additional information
+ * # regarding copyright ownership. The ASF licenses this file
+ * # to you under the Apache License, Version 2.0 (the
+ * # "License"); you may not use this file except in compliance
+ * # with the License. You may obtain a copy of the License at
+ * #
+ * #     http://www.apache.org/licenses/LICENSE-2.0
+ * #
+ * # Unless required by applicable law or agreed to in writing, software
+ * # distributed under the License is distributed on an "AS IS" BASIS,
+ * # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * # See the License for the specific language governing permissions and
+ * # limitations under the License.
+ */
+
+package highlight
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/milvus-io/milvus-proto/go-api/v2/commonpb"
+	"github.com/milvus-io/milvus-proto/go-api/v2/schemapb"
+	"github.com/milvus-io/milvus/internal/util/function/models"
+)
+
+type semanticHighlightProvider interface {
+	highlight(ctx context.Context, query string, texts []string, params map[string]string) ([][]string, error)
+	maxBatch() int
+}
+
+type baseSemanticHighlightProvider struct {
+	batchSize int
+}
+
+func (provider *baseSemanticHighlightProvider) maxBatch() int {
+	return provider.batchSize
+}
+
+type SemanticHighlight struct {
+	fieldIDs []int64
+	provider semanticHighlightProvider
+	queries  []string
+}
+
+const (
+	queryKeyName      string = "queries"
+	inputFieldKeyName string = "input_fields"
+)
+
+func NewSemanticHighlight(collSchema *schemapb.CollectionSchema, params []*commonpb.KeyValuePair, conf map[string]string, extraInfo *models.ModelExtraInfo) (*SemanticHighlight, error) {
```

Comment:
What is diff between `SegmanticHighlighter` and  `SegmanticHighlight`? Can we merge them?

Reply:
SemanticHighlighter is the interface on the proxy, while SemanticHighlight is the concrete implementation.